### ISSUE TITLE
netclass collector fails with permission errors on InfiniBand interfaces

### ISSUE THREAD

**Issue by dswarbrick** [prometheus/procfs#272]
Collector fails with errors such as:
```
prometheus-node-exporter[2255]: time="2020-03-02T17:32:12Z" level=error msg="ERROR: netclass collector failed after 0.011505s: could not get net class info: error obtaining net class info: failed to read file \"/sys/class/net/ib0/create_child\": open /sys/class/net/ib0/create_child: permission denied" source="collector.go:132"
```
As far as I can tell, this is because the collector ranges over all files in the directory, and tries to read their contents if they are a regular file. In the case of InfiniBand interfaces, there are two files which are not world-readable (and in fact not even intended to be read by root):
```
--w------- 1 root root 4096 Mai  7  2019 create_child
--w------- 1 root root 4096 Mär  2 17:41 delete_child
```
For those unfamiliar with InfiniBand, the `create_child` entry is used to create an IB subinterface, with specific P_key (aka partition - similar to an ethernet VLAN). The `delete_child` entry is for removing a subinterface.

The collector would need to perform an additional check besides the existing `f.Mode().IsRegular()` check, or more gracefully handle "permission denied" errors.

**Comment by dswarbrick** [comment]
I'm not sure if this quite qualifies as a regression, but certainly change in how this collector is implemented, introduced by #198, resulted in this bug.

IMHO, I think that blindly reading every file in a sysfs or procfs directory is inherently fragile and potentially even dangerous. Suppose some new kernel introduces a new entry in the directory that produces a massive amount of data when read (think `/proc/kcore`). Granted, such files are normally only readable by root - but some people may (perhaps foolishly) run node_exporter as root. Suppose some buggy driver driver exposes an entry that causes the hardware to malfunction when read (even though we may not be interested in that file for this collector).

Personally, I think it would be a lot safer to explicitly read the files that we are interested in, and not make any assumptions about what else might be in the directory.

**Comment by SuperQ** [comment]
Good catch, I'll get the procfs fix merged as soon as it's ready.
